commands/plugin_bot.go

```diff
@@ -186,6 +186,23 @@ func YAGCommandMiddleware(inner dcmd.RunFunc) dcmd.RunFunc {
 			}
 		}
 
+		// Check if the user can execute the command
+		canExecute, resp, settings, err := yc.checkCanExecuteCommand(data)
+		if err != nil {
+			yc.Logger(data).WithError(err).Error("An error occured while checking if we could run command")
+		}
+
+		parseErr := dcmd.ParseCmdArgs(data)
+		// Stop here if it's autocomplete; we don't need to block it
+		if data.SlashCommandTriggerData.Interaction.Type == discordgo.InteractionApplicationCommandAutocomplete {
```

Comment:
this will panic if data.SlasCommandTriggerData is nil... same thing for method Run in yagcommand.go